### PR TITLE
Test automation allergy

### PR DESCRIPTION
**Pull request by pekkamarjamaki**
Questioning the snake-oil approach to test automation

### CODE REVIEW

File: _posts/2015-07-15-testautomation-allergy.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+layout: post
+title: Test automation allergy
+author: pekkama
+excerpt: Questioning the snake-oil approach to test automation
+---
+
+I am allergic to test automation!
+---------------------------------
+
+That's correct! I get itchy and my eyes redden every time some talks about test automation. I get coarse throat when I see a test automation coverage report. I get twitches when we justify our lack of skills as testers with 95% line coverage. That's a reaction to poorly executed and poorly justified test automation. I am allergic to s***ty test automation. And I think everyone should be!
```
Comment (Pinqvin) [inline _posts/2015-07-15-testautomation-allergy.md]: "I get itchy and my eyes redden every time some talks about test automation." some => someone

File: _posts/2015-07-15-testautomation-allergy.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+This post is about questioning the snake-oil approach to test automation. I have been doing software testing for a number of years in different types of organizations and workplaces. My approach has been skill-driven and holistic, and it is still today. Test automation done well is worth the money invested in it. There has been only few cases where it has been worth the money, though. The best automation is always a supporting act to other testing activities, and the best automation is treated that way.
```
Comment (Pinqvin) [inline _posts/2015-07-15-testautomation-allergy.md]: "My approach has been skill-driven and holistic, and it is still today" Oisko näin parempi: "My approach has been, and still is, skill-driven and holistic."

File: _posts/2015-07-15-testautomation-allergy.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+
+You cannot automate testing!
+----------------------------
+
+Testing isn't a set of steps. By trivializing testing to just execution of steps is plain stupidity! Everyone who thinks like that is essentially in denial, ignorant or a stupid in the head. Testing is a vast set of mental and physical activities. It is executing tests, designing tests, analyzing the results, guessing, trying, learning, understanding, and reporting. More than anything, testing is thinking. Machines don't think, they assert only the stuff we tell them to. We cannot create so comprehensive, complex machine that does the job of a human brain. We can automate something, yes. My suggestion is to understand the limits of test automation and make the best of the stuff it is good at.
```
Comment (Pinqvin) [inline _posts/2015-07-15-testautomation-allergy.md]: "By trivializing testing to just execution of steps is plain stupidity!" => "Trivializing testing to just execution of steps is plain studidity!"

File: _posts/2015-07-15-testautomation-allergy.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+A human cannot test that on one go, but I bet my year’s salary that humans fare better at the task than a machine. Of course contexts vary and there might be occasions where a test automation can test absolutely everything. I (nor anyone whose opinion I value) have ever seen that.
+
+
+The framework of integrity and constant improvement
+----------------------------------------------------
+
+Why the f*** do we want to repeat a test? Why the f*** do we want to repeat a test? Why the f*** do we want to repeat a test? Why the f*** do we want to repeat a test? 
+
+See how silly it is? If you think of it, there are many variables we don't know. We have randomness in our servers, networks, everywhere! Even bloody electrical current isn't stable. So why on earth do we want to try and repeat the situation over and over, if the world we live in is not repeating itself? You might want to answer "to verify that regression doesn't exist". Isn't the point and purpose of regression testing to see if there is something broken? Shouldn't we be searching for that thing that MIGHT be broken?
```
Comment (Pinqvin) [inline _posts/2015-07-15-testautomation-allergy.md]: "If you think of it, there are many variables we don't know." => "If you think about it, there are many variables we don't know."

"Even bloody electrical current isn't stable" => "Even the bloody electrical current isn't stable"

"Shouldn't we be searching for that thing that MIGHT be broken?" => "Shouldn't we be searching for the thing that MIGHT be broken?"

File: _posts/2015-07-15-testautomation-allergy.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+
+Passion for good testing
+------------------------
+
+"Doing your best work includes having the courage to let go of pretty good work that you’ve done before." - [James Bach](http://www.satisfice.com/blog/archives/1343)
+
+In Solita we take pride in having the courage to do things in ways that are perhaps new and daring. We have passion for what we do. We care about people our work affects. 
+
+I want better automation! I don't want to settle for sub-standard test automation where "every test goes red every time something changes even slightly". Toss that s***! Build robust set of checks and take them for what they are: checks. If you want good testing, have a person do it. The person will most likely see areas that require automation to make testing easier. If you are a programmer who writes test automation, be a professional programmer! Use whatever programming practices you use to make the automation code as good as possible. Start simple, keep it simple. If you don't know what to test or how to test, learn to do testing! I bet there are people willing to help you out if you feel stuck in your quest to test intelligently.
```
Comment (Pinqvin) [inline _posts/2015-07-15-testautomation-allergy.md]: "Build robust set of checks and take them for what they are: checks" => "Build a robust set of checks and take them for what they are: checks"

File: _posts/2015-07-15-testautomation-allergy.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
+---
+layout: post
+title: Test automation allergy
+author: pekkama
+excerpt: Questioning the snake-oil approach to test automation
+---
```
Comment (Kauko) [inline _posts/2015-07-15-testautomation-allergy.md]: Tagit puuttuu:
tags:
- testing

File: _posts/2015-07-15-testautomation-allergy.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+I am allergic to test automation!
+---------------------------------
+
+That's correct! I get itchy and my eyes redden every time someome talks about test automation. I get coarse throat when I see a test automation coverage report. I get twitches when we justify our lack of skills as testers with 95% line coverage. That's a reaction to poorly executed and poorly justified test automation. I am allergic to s***ty test automation. And I think everyone should be!
+
+This post is about questioning the snake-oil approach to test automation. I have been doing software testing for a number of years in different types of organizations and workplaces. My approach has been, and still is, skill-driven and holistic. Test automation done well is worth the money invested in it. There has been only few cases where it has been worth the money, though. The best automation is always a supporting act to other testing activities, and the best automation is treated that way.
+
```
Comment (Kauko) [inline _posts/2015-07-15-testautomation-allergy.md]: "Test automation done well is worth the money invested in it. There has been only few cases where it has been worth the money, though." Outo lause. Tarkoitatko jotain tähän suuntaan: "Test automation done well is worth the money invested in in it. Sadly I've only seen a few cases where it has actually been done well, and thus been worth the money."?

Ei välttämättä kovin hyvä tuo mun lause, mutta lukijana tuntuu että alkuperäisessä versiossa on ristiriita, eikä tule selväksi mitä yrität sanoa.

File: _posts/2015-07-15-testautomation-allergy.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+Actor Frank Langella said in a New York Times [article](http://www.nytimes.com/2014/01/05/theater/frank-langella-steps-into-king-lear-at-bam.html?hpw&rref=theater&_r=0)
+
+> “There’s a tendency to value consistency over creativity. You get it, you nail it, you repeat it. I’d rather hang myself. To me, every night within a certain framework — the framework of integrity — you must forget what you did the night before and create it anew every single time you walk out on the stage.”
+
+This is what test automation does: it finds a way that doesn't break the system and then repeats it. As a tester - as a thinking individual - I choose creativity and ideas before repeatability. I try not to do stupid things, but do the best possible testing every time I do testing. I don't fake it, I try to do the most important thing and I try to communicate my work to other people in a way that makes sense. Test automation doesn't strive for perfection, it settles for acceptable, meager existence.
```
Comment (Kauko) [inline _posts/2015-07-15-testautomation-allergy.md]: "This is what test automation does: it finds a way that doesn't break the system and then repeats it." Tää oli hyvä lause, tulee heti mieleen "wall of green" ja siitä saatava valheellinen hyvä mieli. Mulla oli vaikea ymmärtää mitä haet takaa ennen quotea olevilla kappaleilla, miettisin niitä vielä uusiksi.

File: _posts/2015-07-15-testautomation-allergy.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+Passion for good testing
+------------------------
+
+> "Doing your best work includes having the courage to let go of pretty good work that you’ve done before." - [James Bach](http://www.satisfice.com/blog/archives/1343)
+
+In Solita we take pride in having the courage to do things in ways that are perhaps new and daring. We have passion for what we do. We care about people our work affects. 
+
+I want better automation! I don't want to settle for sub-standard test automation where "every test goes red every time something changes even slightly". Toss that s***! Build a robust set of checks and take them for what they are: checks If you want good testing, have a person do it. The person will most likely see areas that require automation to make testing easier. If you are a programmer who writes test automation, be a professional programmer! Use whatever programming practices you use to make the automation code as good as possible. Start simple, keep it simple. If you don't know what to test or how to test, learn to do testing! I bet there are people willing to help you out if you feel stuck in your quest to test intelligently.
+
```
Comment (Kauko) [inline _posts/2015-07-15-testautomation-allergy.md]: "- for what they are: checks If you want -" <- piste puuttu checks jälkeen.

**Comment by Rinorragi** [comment]
Closed after confirmation from author

**Pull request closed without merging**